overlord/snapstate/hooks/hooks.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hooks
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/i18n/dumb"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+func init() {
+	snapstate.InstallHookSetup = InstallHookSetup
+	snapstate.RemoveHookSetup = RemoveHookSetup
+}
+
+func InstallHookSetup(st *state.State, snapName string) *state.Task {
+	hooksup := &hookstate.HookSetup{
+		Snap:     snapName,
+		Hook:     "install",
+		Optional: true,
+	}
+
+	summary := fmt.Sprintf(i18n.G("Install hook of snap %q"), hooksup.Snap)
```

Comment:
Similar problem here. We need the task summary to read like a task. Looking at the existing configure hook wording, this should be: 

_"Run install hook of %q snap"_

Same for remove below.